lib/new_relic/agent/transaction.rb

```diff
@@ -261,6 +261,30 @@ def initialize(category, options)
         end
       end
 
+      def parent_thread_id
+        Thread.current.nr_parent_thread_id if Thread.current.respond_to?(:nr_parent_thread_id)
+      end
+
+      def current_segment
+        # if this thread id has a current segment use that
+        current_thread_id = Thread.current.object_id
+        return current_segment_by_thread[current_thread_id] if current_segment_by_thread[current_thread_id]
+
+        # if this thread id does not have a current segment, this is where we'll need thread parent
+        return current_segment_by_thread[parent_thread_id] if current_segment_by_thread[parent_thread_id]
+
+        # if both of those fail, fall back to starting thread current segment
+        return current_segment_by_thread[@starting_thread_id]
```

Comment:
No need for the explicit `return` on the default case here.

Reply:
True, i just liked the symmetry with the lines above it when it has the return. I can totally remove it though, since that is the standard way to do it.